rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/AppVersionRevision.scala

```diff
@@ -34,21 +34,24 @@ final class AppVersionRevision(implicit system: ExtendedActorSystem) extends Ext
   def getRevision(): Future[Version] = versionPromise.future
 
   def start(): Unit = {
-    if (k8sSettings.podName.isEmpty) {
-      log.error(
-        "Not able to read the app version from the revision of the current ReplicaSet. Reason:" +
-        "No configuration found to extract the pod name from. " +
-        s"Be sure to provide the pod name with `$configPath.pod-name` " +
-        "or by setting ENV variable `KUBERNETES_POD_NAME`.")
-    } else {
-      if (isInitialized.compareAndSet(false, true)) {
+    if (isInitialized.compareAndSet(false, true)) {
+      if (k8sSettings.podName.isEmpty) {
+        val msg = "Not able to read the app version from the revision of the current ReplicaSet. Reason:" +
+          "No configuration found to extract the pod name from. " +
+          s"Be sure to provide the pod name with `$configPath.pod-name` " +
+          "or by setting ENV variable `KUBERNETES_POD_NAME`."
+        log.error(msg)
+        versionPromise.failure(new IllegalStateException(msg))
```

Comment:
If podname is not set, there is nothing else we can do, but if we try to later use `getRevision` we get a Future that never completes.

Reply:
Hmm, maybe we should not return an `IllegalStateException`, because it's not illegal. It's just missing info causing a degraded state.